resen/Resen.py

```diff
@@ -642,12 +652,16 @@ def create_container(self,**input_kwargs):
             create_kwargs['volumes'][key] = temp
 
         # check if we have image, if not, pull it
-        reps = [x.attrs['RepoTags'] for x in self.docker.images.list()]
-        if not image_name in reps:
+        local_image_ids = [x.id for x in self.docker.images.list()]
+        if image_id not in local_image_ids:
+            # get the repo digest hash
+            remote_repodigest = self.docker.images.get_registry_data(image_name).id
+            # verify that the repo has the digest we expect
+            assert image_repodigest == remote_repodigest, \
+                    'Trying to retrieve an image with different digest.'
             print("Pulling image: %s" % image_name)
             print("   This may take some time...")
-            repo, tag = image_name.split(':')
-            self.docker.images.pull(repo,tag=tag)
+            self.docker.images.pull(image_name)
```

Comment:
We need to pull the image using a different name. We want to pull the image using the digest now. So instead of `image_name`, we should be using something like:

    pull_name = 'docker.io/%s/%s@%s' % (org,repo,digest)
    self.docker.images.pull(pull_name)

This will probably require building the pull name in the `add_image` method, similar to how the `image` parameter is currently built.

Reply:
The `pull_name` will be created  in `add_image` method.